pxr/usd/usdShade/validators.cpp

```diff
@@ -590,6 +590,157 @@ _SubsetsMaterialBindFamily(const UsdPrim& usdPrim)
+                               usdPrim.GetPath()) };
+    if (implementationSource != UsdShadeTokens->id)
+    {
+        errors.emplace_back(
+            UsdShadeValidationErrorNameTokens->nonIdImplementationSource,
+            UsdValidationErrorType::Error,
+            primErrorSite,
+            TfStringPrintf(
+            "Shader <%s> has non-id implementation "
+                "source '%s'.",
+                usdPrim.GetPath().GetText(),
+                implementationSource.GetText())
+        );
+    }
+    TfToken shaderId;
+    const bool gotShaderId = shaderPrim.GetShaderId(&shaderId);
+
+    const std::vector<TfToken> validShaderIds = {
+        TfToken("UsdPreviewSurface"),
+        TfToken("UsdUVTexture"),
+        TfToken("UsdTransform2d"),
+        TfToken("UsdPrimvarReader")};
+
+    auto isValidShaderId = [&]()
+    {
+        bool isKnownShaderName = std::find(validShaderIds.begin(),
+            validShaderIds.end(), shaderId) != validShaderIds.end();
+        bool isKnownShaderPrefix = TfStringStartsWith(shaderId.GetText(),
+            "UsdPrimvarReader") ||
+                TfStringStartsWith(shaderId.GetText(), "ND_");
+
+        return isKnownShaderName || isKnownShaderPrefix;
+    };
+
+    if (!gotShaderId)
+    {
+        errors.emplace_back(
+            UsdShadeValidationErrorNameTokens->invalidShaderId,
+            UsdValidationErrorType::Error,
+            primErrorSite,
+            TfStringPrintf(
+            "Shader <%s> has unsupported info:id 'None'.",
+                usdPrim.GetPath().GetText())
+        );
+    }
+    else if(!isValidShaderId())
+    {
+        errors.emplace_back(
+            UsdShadeValidationErrorNameTokens->invalidShaderId,
+            UsdValidationErrorType::Error,
+            primErrorSite,
+            TfStringPrintf(
+            "Shader <%s> has unsupported info:id '%s'.",
+                usdPrim.GetPath().GetText(),
+                shaderId.GetText())
+        );
+    }
+
+    const std::vector<UsdShadeInput> shaderInputs = shaderPrim.GetInputs();
+
+    // Check shader input connections
+    for(const UsdShadeInput &shaderInput: shaderInputs)
+    {
+        SdfPathVector sources;
+        shaderInput.GetAttr().GetConnections(&sources);
+
+        // If an input has one or more connections, ensure that the
+        // connections are valid.
+        if (sources.empty())
+        {
+            continue;
+        }
+
+        if (sources.size() > 1)
+        {
+            errors.emplace_back(
+                UsdShadeValidationErrorNameTokens->multipleConnectionSources,
+                UsdValidationErrorType::Error,
+                primErrorSite,
+                TfStringPrintf(
+                "Shader input <%s> has %lu connection "
+                        "sources, but only one is allowed.",
+                    shaderInput.GetAttr().GetPath().GetText(),
+                    sources.size())
+            );
+            continue;
```

Comment:
@tallytalwar one note here, I noticed this was not present in the other validator [here](https://github.com/PixarAnimationStudios/OpenUSD/blob/release/pxr/usd/usdUtils/complianceChecker.py#L871) however:
1. We do get a warning below when calling `GetConnectedSource` since with multiple sources we should call `GetConnectedSources`
2. It appears `GetConnectedSource` gets the first connected source, so the complianceChecker code would not be validating issues for other connected sources

Happy to change this and loop through the connected sources if you think it's better to add that logic in, it made sense to me to stop checking since no matter what the connections were, it seems invalid to have more than one. Let me know if that's incorrect.